homeassistant/components/huum/config_flow.py

```diff
@@ -0,0 +1,63 @@
+"""Config flow for huum integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from huum.exceptions import Forbidden, NotAuthenticated
+from huum.huum import Huum
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class HuumConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for huum."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                huum_handler = Huum(
+                    user_input[CONF_USERNAME],
+                    user_input[CONF_PASSWORD],
+                    session=async_get_clientsession(self.hass),
+                )
+                await huum_handler.status()
+            except (Forbidden, NotAuthenticated):
+                # Most likely Forbidden as that is what is returned from `.status()` with bad creds
+                _LOGGER.error("Could not log in to Huum with given credentials")
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unknown error")
+                errors["base"] = "unknown"
+            else:
+                self._async_abort_entries_match(
+                    {CONF_USERNAME: user_input[CONF_USERNAME]}
```

Comment:
I suppose there is no better way to identify the sauna than the username? A single user can have only a single sauna, is that correct?

Reply:
I will refer you to this answer https://github.com/home-assistant/core/pull/106420#discussion_r1437024134

TL;DR A better identifier exists, but we don't have "explicit permission" to use that API as it is not under the same API prefix, it uses a different auth method and there is no documentation other than my MiTM testing for the API 😄  The same goes for the sauna id, the user id and settings temperature without turning on the sauna.

Reply:
Okay, I think this is then good enough for the time being! 👍